File: scripts/firehose.py
#!/usr/bin/env python
# pip install python-socketio

import socketio
import datetime

sio = socketio.Client()

# setting optional start_date parameter
now = datetime.datetime.utcnow()

sio.connect(
    f'http://localhost:8080?start_date=dd{now}',
    namespaces=['/reports/firehose'],
)

@sio.event(namespace='/reports/firehose')
def status_report(data):
    print(data)

@sio.event(namespace='/reports/firehose')
def exception(data):
    print(data)

File: scripts/firehose.js
// npm install socket.io-client
const io = require('socket.io-client');

// setting optional start_date parameter
const now = new Date();
const options = { query: `start_date=${now.toISOString()}` };

const socket = io('https://api.status-api.com/reports/firehose', options);

socket.on('status_report', (data) => {
    console.log(data);
});
socket.on('exception', (data) => {
    console.error(data);
});